src/Microsoft.Extensions.Configuration.DockerSecrets/DockerSecretsConfigurationProvider.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Extensions.FileProviders;
+
+namespace Microsoft.Extensions.Configuration.DockerSecrets
+{
+    /// <summary>
+    /// An docker secrets based <see cref="ConfigurationProvider"/>.
+    /// </summary>
+    public class DockerSecretsConfigurationProvider : ConfigurationProvider
+    {
+        DockerSecretsConfigurationSource Source { get; set; }
+
+        /// <summary>
+        /// Initializes a new instance.
+        /// </summary>
+        /// <param name="source">The settings.</param>
+        public DockerSecretsConfigurationProvider(DockerSecretsConfigurationSource source)
+        {
+            Source = source ?? throw new ArgumentNullException(nameof(source));
+        }
+
+        /// <summary>
+        /// Loads the docker secrets.
+        /// </summary>
+        public override void Load()
+        {
+            Data = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+
+            if (Source.FileProvider == null)
+            {
+                if (Directory.Exists(Source.SecretsDirectory))
+                {
+                    Source.FileProvider = new PhysicalFileProvider(Source.SecretsDirectory);
+                }
+                else if (Source.Optional)
+                {
+                    return;
+                }
+                else
+                {
+                    throw new FileNotFoundException("Secrets directory doesn't exist and is not optional.");
+                }
+            }
+
+            var secretsDir = Source.FileProvider.GetDirectoryContents("/");
+            if (!secretsDir.Exists && !Source.Optional)
+            {
+                throw new FileNotFoundException("Secrets directory doesn't exist and is not optional.");
```

Comment:
message should include the word `Docker` to avoid confusing with `Microsoft.Extensions.Configuration.UserSecrets`. `Docker Secrets` instead of `Secrets`